composer/callbacks/checkpoint_saver.py

```diff
@@ -319,16 +333,19 @@ def _save_checkpoint(self, state: State, logger: Logger):
 
         # save the checkpoint to the filename
         filename_with_placeholders = self.filename.format(state, is_deepspeed, keep_placeholders=True)
+        save_filename = checkpoint.get_save_filename(state, filename_with_placeholders)
+        self.all_saved_checkpoints_to_timestamp[save_filename] = state.timestamp
 
-        saved_path = checkpoint.save_checkpoint(
+        saved_path = checkpoint._save_checkpoint(
```

Comment:
Split this up in checkpoint.save_checkpoint so that the callback could inject the current checkpoint into the state otherwise it's always one-off. Looks like `checkpoint.save_checkpoint` is heavily documented so didn't want to make it backwards breaking